tests/test_dsr.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+import pytest
+from fastapi.testclient import TestClient
+
+from datahub import data as dt
+from datahub.main import app
+
+client = TestClient(app)
+client.headers["Content-Type"] = "application/json"
+
+
+@pytest.fixture(autouse=True)
+def reset_dsr_data():
+    """Pytest Fixture for resetting DSR data global variable."""
+    dt.opal_df = []
```

Comment:
Ignoring the performance issues, all looks good to me. I wasn't able test the new endpoint manually with Postman as I couldn't get it working with a separate .json file. Might be an area I should try and familiarise myself with.

Only thing I caught was the `reset_dsr_data` fixture accessing the Opal global variable instead of the DSR one.

```suggestion
    dt.dsr_data = []
```

The integrated tests are also failing as `httpx` isn't included in the dev requirements. They are included in https://github.com/ImperialCollegeLondon/gridlington-datahub/pull/69 so might need to push that first then update this branch.